### ISSUE TITLE
How to output HTML in a literal block alongside rest generated HTML

### ISSUE THREAD

**Issue by lakhman** [sphinx-doc/sphinx#3995]
Subject: Output HTML generated by Sphinx into a ``literal_block`` alongside the normal regular output.

### Problem
I've written a directive which takes some rest text and outputs HTML (like usual), but I also want to output that same generated HTML source into a ``literal_block`` next to the original element. I can use ``nodes.raw`` to output the *Sphinx XML representation* (like below), but I want to output the *generated HTML* - not the *Sphinx XML representation* as held by Sphinx doc (which it what it does).

e.g (currently): ``<container><paragraph>Example content</paragraph></container>`` ← This is what is outputted into the literal block.
e.g (expected): ``<pre><div><p>Example content</p></div></pre>`` ← I want this in the literal block

How can I do this? I also tried doing this in the ``post-transform`` phase, but I get the same result (it outputs the Sphinx XML representation, not the HTML generated).

#### Procedure to reproduce the problem
```
.. container::

  Example content
```

> ↓ - Should render into 2 elements, the container and literal_block... (via my custom directive or post-transform) ↓

```html
<div>
<p>Example Content</p>
</div>
<!-- Note: The `container` is ALSO within a `literal_block` (a sibling to it) -->
<pre>
  <div>
    <p>Example Content</p>
  </div>
</pre>
```

My directive looks kind of like this:

```python
# class codeblock_example(nodes.General, nodes.Element): pass

# Parse our original content into a node (like usual)
parsed = codeblock_example(code, **self.options)
self.state.nested_parse(self.content, self.content_offset, parsed)
# `parsed` contains the XML representation, I want to put it into a `literal_block` (as HTML)

# Add that parsed content into a literal block next to the original
literal = nodes.literal_block(parsed)
literal.document = self.state.document
literal['language'] = 'html'
literal.append(nodes.raw(parsed, parsed))

# I tried using `publish_parts`, but this doesn't take into account my custom extension.
# https://stackoverflow.com/questions/6654519/parsing-restructuredtext-into-html

# .. append both into a container (the HTML to render, and the source to view inside a <pre>)
```

*So basically*, I want to output the source generated by sphinx (as usual) - but also next to it, the actual HTML source inside a ``literal_block`` - so we can see exactly what was generated (in the browser).

Any help/ideas would be much appreciated. 👍

**Comment by lakhman** [comment]
I've found a workaround using javascript to do this once the browser has loaded.
